app/src/main/java/com/kodeco/android/coordplot/components/PlotSurface.kt

```diff
@@ -0,0 +1,97 @@
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.kodeco.android.coordplot.R
+import com.kodeco.android.coordplot.ui.theme.CoordPlotTheme
+import com.kodeco.android.coordplot.utils.toIntPercentage
+
+@Composable
+fun PlotSurface(screenOrientation: Int) {
+    var xFloatValue: Float by rememberSaveable { mutableStateOf(0.5f) }
+    var yFloatValue: Float by rememberSaveable { mutableStateOf(0.5f) }
+
+    if (screenOrientation == Configuration.ORIENTATION_PORTRAIT) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center,
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(16.dp)
+        ) {
+            Map(xPercent = xFloatValue, yPercent = yFloatValue)
+            MapSlider(
+                floatValue = xFloatValue,
+                onValueChanged = { xFloatValue = it },
+                text = stringResource(
+                    id = R.string.slider_axis_value,
+                    stringResource(R.string.x_label),
+                    xFloatValue.toIntPercentage()
+                )
+            )
+            MapSlider(
+                floatValue = yFloatValue,
+                onValueChanged = { yFloatValue = it },
+                text = stringResource(
+                    id = R.string.slider_axis_value,
+                    stringResource(R.string.y_label),
+                    yFloatValue.toIntPercentage()
+                )
+            )
+        }
+    } else {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Center,
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(16.dp)
+        ) {
+            Map(xPercent = xFloatValue, yPercent = yFloatValue)
+            Column {
+                MapSlider(
+                    floatValue = xFloatValue,
+                    onValueChanged = { xFloatValue = it },
+                    text = stringResource(
+                        id = R.string.slider_axis_value,
+                        stringResource(R.string.x_label),
+                        xFloatValue.toIntPercentage()
+                    )
+                )
+                MapSlider(
+                    floatValue = yFloatValue,
+                    onValueChanged = { yFloatValue = it },
+                    text = stringResource(
+                        id = R.string.slider_axis_value,
+                        stringResource(R.string.y_label),
+                        yFloatValue.toIntPercentage()
+                    )
+                )
+            }
+        }
+    }
```

Comment:
I could not think for a better solution at the moment. Would appreciate any suggestions

Reply:
Your solution works fine for both portrait and landscape. 

Another option could be to use `BoxWithConstraints` as mentioned in the official doc (https://developer.android.com/jetpack/compose/layouts/adaptive). You can also check out the [sample apps](https://developer.android.com/jetpack/compose/layouts/adaptive#sample_apps)  to see how they handle layouts based on screen size.